addons/website/tests/test_res_users.py

```diff
@@ -110,3 +112,19 @@ def test_archive_company_linked_to_website(self):
         # The company cannot be archived because it has a website linked to it
         with self.assertRaises(ValidationError):
             company.action_archive()
+
+    def test_user_become_internal(self):
+        # This tests if the website_id is correctly removed when a user becomes
+        # an internal user.
+        website = self.env['website'].create({'name': "Awesome Website"})
+        # Permit uninvited signup for portal users.
+        website.auth_signup_uninvited = 'b2c'
+        # Disable cross-website for portal users.
+        website.specific_user_account = True
+        user = self._create_user_via_website(website, 'Portal_User')
+        self.assertEqual(user.website_id, website)
+        self.assertTrue(user._is_portal())
+        internal_group = self.env.ref('base.group_user')
+        portal_group = self.env.ref('base.group_portal')
+        with self.assertRaises(ValidationError):
+            user.groups_id = [Command.link(internal_group.id), Command.unlink(portal_group.id)]
```

Comment:
Nitpick:
```suggestion
        with self.assertRaises(ValidationError):
            user.groups_id = [
                Command.link(self.env.ref('base.group_user').id),
                Command.unlink(self.env.ref('base.group_portal').id),
            ]
```
Variable doesn't bring much, + make code more aerated + easy to extend the list in case we need, but ignore if you pref

Reply:
You ignored it right? Just to be sure since not changed but resolved

Reply:
Yes I already changed it :eyes: 
![image](https://github.com/odoo/odoo/assets/78849981/e563eb05-4ce7-48a9-aedb-4a10f6cb0107)